### PR TITLE
initializeClients function to return struct of the clients. 

### LINKED ISSUE #531: CleanUp the kanctl code to return a struct of clients
In current implementation the [initializeClients](https://github.com/kanisterio/kanister/blob/master/pkg/kanctl/kanctl.go#L80)  function returns four variable instead of that we should rather return a struct of the clients.

### PR DESCRIPTION
**Pull request by harshitasao**
Signed-off-by: harshitasao <harshitasao@gmail.com>

## Change Overview

Made the initializeClients function return a struct of clients so that when we try to return any other client, the list does not continue to grow.
## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #531 

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/kanctl/kanctl.go [hunk 1]
```diff
@@ -73,24 +79,30 @@ func resolveNamespace(cmd *cobra.Command) (string, error) {
 	return kube.ConfigNamespace()
 }
 
-func initializeClients() (kubernetes.Interface, versioned.Interface, osversioned.Interface, error) {
+func initializeClients() (*Clients, error) {
+	nilClient := Clients{}
 	config, err := kube.LoadConfig()
 	if err != nil {
-		return nil, nil, nil, err
+		return &nilClient, err
```
Comment (ihcsim) [inline pkg/kanctl/kanctl.go:86]: Why not just `return nil, err`? Same for line 90, 95, 100.

File: pkg/kanctl/kanctl.go [hunk 2]
```diff
@@ -39,6 +39,12 @@ var (
 	Verbose bool
 )
 
+type Clients struct {
+	KubeClient kubernetes.Interface
+	CrdClient  versioned.Interface
```
Comment (ihcsim) [inline pkg/kanctl/kanctl.go:44]: Can we name this `KanisterClient`, since it calls the Kanister API?

File: pkg/kanctl/kanctl.go [hunk 3]
```diff
@@ -39,6 +39,12 @@ var (
...
+	OsClient   osversioned.Interface
```
Comment (ihcsim) [inline pkg/kanctl/kanctl.go:45]: s/OsClient/OSClient

following the Golang convention on initialism: https://github.com/golang/go/wiki/CodeReviewComments#initialisms

**Comment by ihcsim** [comment]
@viveksinghggits PTAL.

File: pkg/kanctl/actionset.go [hunk 4]
```diff
@@ -105,24 +105,24 @@ func newActionSetCmd() *cobra.Command {
 }
 
 func initializeAndPerform(cmd *cobra.Command, args []string) error {
-	cli, crCli, osCli, err := initializeClients()
+	clients, err := initializeClients()
 	if err != nil {
 		return err
 	}
-	params, err := extractPerformParams(cmd, args, cli, osCli)
+	params, err := extractPerformParams(cmd, args, clients.KubeClient, clients.OsClient)
```
Comment (viveksinghggits) [inline pkg/kanctl/actionset.go:112]: @ihcsim would it be a good idea if we directly pass `clients` here?

File: pkg/kanctl/kanctl.go [hunk 5]
```diff
@@ -39,6 +39,12 @@ var (
 	Verbose bool
 )
 
+type Clients struct {
+	KanisterClient kubernetes.Interface
+	CrdClient      versioned.Interface
+	OSClient       osversioned.Interface
+}
```
Comment (PrasadG193) [inline pkg/kanctl/kanctl.go:46]:
```suggestion
type Clients struct {
	KubeClient kubernetes.Interface
	KanisterClient      versioned.Interface
	OSClient       osversioned.Interface
}
```
Let's rename CrdClient to KanisterClient

Comment (PrasadG193) [reply pkg/kanctl/kanctl.go:46]: https://github.com/kanisterio/kanister/pull/1666#discussion_r992614443
I think there is confusion. In this comment, Ivan suggested renaming CrdClient not KubeClient :)

Comment (harshitasao) [reply pkg/kanctl/kanctl.go:46]: Understood, thanks!

**Comment by viveksinghggits** [comment]
LGTM overall. @harshitasao will be it possible for you to add any tests that you have performed after the changes. I would recommend, building the kanctl binary and then creating the profile/actionset object would be good idea. You can follow any example from here after building the binary https://github.com/kanisterio/kanister/tree/master/examples/mysql.

**Pull request closed without merging**